tests.py

```diff
@@ -12,40 +12,38 @@
 import os
 import sys
 import time
+import pytest
+from matplotlib import default_test_modules
 
 import matplotlib
 matplotlib.use('agg')
 
-import nose
-from matplotlib import default_test_modules
-
 
 def run(extra_args):
-    from nose.plugins import multiprocess
-
-    env = matplotlib._get_nose_env()
-
     matplotlib._init_tests()
 
-    # Nose doesn't automatically instantiate all of the plugins in the
-    # child processes, so we have to provide the multiprocess plugin with
-    # a list.
-    plugins = matplotlib._get_extra_test_plugins()
-    multiprocess._instantiate_plugins = plugins
-
-    nose.main(addplugins=[x() for x in plugins],
-              defaultTest=default_test_modules,
-              argv=sys.argv + extra_args,
-              env=env)
+    argv = sys.argv + extra_args
+    # pytest.main(['--pyargs', '--cov=matplotlib'] + default_test_modules)
+    print(argv + ['--pyargs'] + default_test_modules)
+
+    # if a specific test module is in argv, run that instead of all tests
+    for argument in sys.argv:
+        if argument in default_test_modules:
+            # might need to insert a "--pyargs" into argv before passing
+            # to pytest.main
+            return pytest.main(argv)
+    else:
+        pytest.main(['--traceconfig'])
+        return pytest.main(argv + ['--pyargs'] + default_test_modules +
+                           ['--ignore=site-packages/numpy/testing/.'])
```

Comment:
This is making strong assumptions about how numpy is installed, is there a way to ask numpy where it is insalled?